identity_stardust/src/client/client_ext.rs

```diff
@@ -0,0 +1,616 @@
+  /// - Returns a [`NotFound`](iota_client::Error::NotFound) error if the associated Alias Output wasn't found.
+  async fn resolve_alias_output(&self, did: &StardustDID) -> Result<AliasOutput> {
+    let network_hrp: String = get_network_hrp(self.client()).await?;
+
+    if did.network_str() != network_hrp.as_str() {
+      return Err(Error::NetworkMismatch {
+        expected: did.network_str().to_owned(),
+        actual: network_hrp,
+      });
+    }
+
+    resolve_alias_output(self.client(), did)
+      .await
+      .map(|(_, _, alias_output)| alias_output)
+  }
+
+  /// Returns the network name of the connected node, which is the
+  /// BECH32 human-readable part (HRP) of the network.
+  ///
+  /// For the IOTA main network this is `iota` and for the Shimmer network it is `smr`.
+  async fn network_name(&self) -> Result<NetworkName> {
+    get_network_hrp(self.client()).await.and_then(NetworkName::try_from)
+  }
+}
+
+impl StardustClientExt for Client {
+  fn client(&self) -> &Client {
+    self
+  }
+}
+
+impl StardustClientExt for &Client {
+  fn client(&self) -> &Client {
+    self
+  }
+}
+
+/// Publishes an `alias_output`.
+/// Returns the block that the output was included in.
+async fn publish_output(client: &Client, secret_manager: &SecretManager, alias_output: AliasOutput) -> Result<Block> {
+  let block: Block = client
+    .block()
+    .with_secret_manager(secret_manager)
+    .with_outputs(vec![alias_output.into()])
+    .map_err(Error::DIDUpdateError)?
+    .finish()
+    .await
+    .map_err(Error::DIDUpdateError)?;
+
+  let _ = client
+    .retry_until_included(&block.id(), None, None)
+    .await
+    .map_err(Error::DIDUpdateError)?;
+
+  Ok(block)
+}
+
+/// Get the BECH32 HRP from the client's network.
+async fn get_network_hrp(client: &Client) -> Result<String> {
+  client
+    .get_network_info()
+    .await
+    .map_err(Error::ResolutionError)?
+    .bech32_hrp
+    .ok_or_else(|| Error::InvalidNetworkName("".to_owned()))
+}
+
+/// Returns all DID documents of the Alias Outputs contained in the payload's transaction, if any.
+async fn documents_from_block(client: &Client, block: &Block) -> Result<Vec<StardustDocument>> {
+  let network_hrp: String = get_network_hrp(client).await?;
+  let mut documents = Vec::new();
+
+  if let Some(Payload::Transaction(tx_payload)) = block.payload() {
+    let TransactionEssence::Regular(regular) = tx_payload.essence();
+
+    for (index, output) in regular.outputs().iter().enumerate() {
+      if let Output::Alias(alias_output) = output {
+        let alias_id = if alias_output.alias_id().is_null() {
+          AliasId::from(
+            OutputId::new(
+              tx_payload.id(),
+              index.try_into().expect("the output count should not exceed u16"),
+            )
+            .expect("the index of any output in a RegularTransactionEssence should be a valid OutputIndex"),
```

Comment:
I can't imagine these invariants being broken but we should map them to errors in case the protocol changes and breaks them.